mlir/lib/Bindings/Python/Pass.cpp

```diff
@@ -76,14 +76,16 @@ void mlir::python::populatePassManagerSubmodule(py::module &m) {
           "enable_ir_printing",
           [](PyPassManager &passManager, bool printBeforeAll,
              bool printAfterAll, bool printModuleScope, bool printAfterChange,
-             bool printAfterFailure) {
+             bool printAfterFailure, std::string treePrintingPath) {
```

Comment:
You can go `std::optional<string>` here and then `"treePrintingPath"_a = py::none()` below and the arg will be `nullopt` when someone doesn't pass it.

Reply:
Why is this better though? The empty string works through API and is handled on the C++ side as disabling the feature.

Reply:
Because `std::optional` is designed to indicate an optional thing and pybind is also designed to be compatible with this (and optionality is mapped to `None` in python, not empty string).

Reply:
But how does it translate to a user?

Reply:
> superficial stylistic argument here

https://discourse.llvm.org/t/maybe-unused-vs-attribute-unused-vs-void-var-for-assert-variables/64212/30